addons/mail/static/src/models/thread_view/thread_view.js

```diff
@@ -201,6 +202,13 @@ function factory(dependencies) {
         isPreparingLoading: attr({
             default: false,
         }),
+        /**
+         * Determines whether `this` would like its `threadCache` to refresh its
+         * messages.
+         */
+        isRequestingCacheRefresh: attr({
+            default: true,
+        }),
```

Comment:
Using this field makes things more cumbersome than it should.
This put concern of cache in views, which makes this out-of-place for cache to depend on its viewers with this field.
Refresh of caches being viewed should be handled immediately to the cache, because this is purely a cache concern.